Initial draft of mission statement of the TLA+ Foundation (#1)
tlaplus/foundation: README.md

# TLA+ Foundation

# 1. Goals of the Foundation

> The TLA+ Foundation (TLAF) is an independent, non-profit organization dedicated to fostering the adoption of the TLA+ specification language in industry, academia, and education, guided by the overall goal to advance the practice of mathematically thinking in software engineering.  The mission of TLAF is active support and funding of the development of the TLA+ tools to broaden their use by practitioners.  If not directly funding work, TLAF creates synergies whenever possible.

Reviewer: I think this is a narrow statement. The Foundation should not be just about the TLA+ tools. It should be about promoting use of TLA+, forming TLA+ communities as well. 
I think maybe we should just state that the mission is to broaden TLA+ use by practitioners, and expand on the means to that end (expanding tool support, promoting/forming communities) separately.

> The TLA+ Foundation (TLAF) is an independent, non-profit organization dedicated to fostering the adoption of the TLA+ specification language in industry, academia, and education, guided by the overall goal to advance the practice of mathematically thinking in software engineering.  The mission of TLAF is active support and funding of the development of the TLA+ tools to broaden their use by practitioners.  If not directly funding work, TLAF creates synergies whenever possible.

Reviewer 2: "mathematically thinking in software engineering" -> "mathematical thinking in software engineering"

> The TLA+ Foundation (TLAF) is an independent, non-profit organization dedicated to fostering the adoption of the TLA+ specification language in industry, academia, and education, guided by the overall goal to advance the practice of mathematically thinking in software engineering.  The mission of TLAF is active support and funding of the development of the TLA+ tools to broaden their use by practitioners.  If not directly funding work, TLAF creates synergies whenever possible.

Reviewer 2: ```suggestion
The TLA+ Foundation (TLAF) is an independent, non-profit organization dedicated to fostering the adoption of the TLA+ specification language in industry, academia, and education, guided by the overall goal of advancing the practice of mathematical thinking in software engineering. The TLAF funds the development of TLA+ tools with the aim to broaden their use by practitioners, and facilitates a thriving TLA+ community through organization of conferences and forums.
```
Retooled grammar a bit. I don't know what the last part about synergies means.

> The TLA+ Foundation (TLAF) is an independent, non-profit organization dedicated to fostering the adoption of the TLA+ specification language in industry, academia, and education, guided by the overall goal of advancing the practice of mathematical thinking in software engineering. The TLAF funds development related to TLA+ to broaden its use by practitioners and facilitate a thriving TLA+ community.  If not actively involved in work, the TLAF aims to create synergies among all community players.

Reviewer 3: Most people now associate "mathematical thinking" with category theory, haskell, and typed FP. This could lead to either confusion or pressure to more in a more CT direction. I don't have a solution for this, just raising it as a possible issue.

> The TLA+ Foundation is also responsible for the evolution of the TLA+ specification language that guarantees long-term consistency.

Reviewer 2: ```suggestion
The TLA+ Foundation is also responsible for the evolution of the TLA+ specification language itself, guaranteeing long-term consistency.
```

# 2. The values of the Foundation

> How we pursue the goals of HF is just as important as what the goals are. These core principles will guide the actions of the Foundation:

Reviewer 2: ```suggestion
How we pursue the goals of the TLAF is just as important as what the goals are. These core principles will guide the actions of the Foundation:
```

> * **Open source**. The TLA+ foundation will embrace the open-source ethos wholeheartedly. Whenever TLAF develops or sponsors the development of tools and infrastructure, it will all be open source under a permissive license.  This is not limited to the end result but also applies to how work is organized and carried out; the community at large has to be able to participate at each step of the journey.

Reviewer 2: ```suggestion
* **Open source**. The TLAF will embrace the open-source ethos wholeheartedly. Whenever the TLAF develops or sponsors the development of tools and infrastructure, it will all be open-source. This is not limited to the end result but also applies to how work is organized and carried out; the community at large has to be able to participate at each step of the journey.
```
Do we have to commit to permissive licenses (MIT, BSD, etc. - not copyleft) in the charter? It can just be our operating practice without being codified.

Author: Committing to permissive licenses signals member companies long-term stability.

Why would we want a copyleft (GPL, ...) license in the first place?  From my experience, it makes things harder for most adopters.

Author: What does the rest of the group think about licenses?

Reviewer 4: @lemmy I really just have a strong instinctive aversion to copyleft based on my experience working in open source, working with large bureaucratic companies that are adopting open source software, and speaking with attorneys specializing in open source software licensing. I think the question we’d need to answer first is: what benefit would we get from copyleft? I’m concerned it’s a risk without an obvious reward, but maybe someone here can justify it for us.

We’ve had discussions with attorneys on this topic at ONF in the past. If we think we need to investigate this further, I may be able to track down the recordings of those discussions and see if there’s any more helpful context in them.

> * **Empowering the community**. While the TLAF raises resources to fund development work, it does not compete with volunteer efforts from industry or academia.  The main focus is to address the tragedy of the commons to enable the success of others. The TLAF will boost, celebrate, and coordinate the contributions of volunteers, not supplant or replace them.

Reviewer 2: ```suggestion
* **Empowering the community**. While the TLAF raises resources to fund development work, it does not compete with volunteer efforts from industry or academia. The main focus is to address the tragedy of the commons to enable the success of others. The TLAF will boost, celebrate, and coordinate the contributions of volunteers, not supplant or replace them.
```

* **Transparent**. All formal decision-making will be publicly accessible. We also strive to share ideas early. This is to enable asynchronous communication and collaboration. Only certain categories of sensitive information (e.g., financial and matters concerning particular individuals) may be kept confidential.

> * **Open, friendly, and diverse**. `We strive to make sure that all lifeforms are welcome, that all can contribute, and that it can all be done in a way that empowers our community members.

Reviewer 3: Trailing \` here. The "lifeforms" thing strikes me as odd.

Reviewer 4: I like the spirit of this, but I think the phrasing could be a bit better… just not sure what exactly that would look like yet
